src/chronify/store.py

```diff
@@ -138,62 +399,464 @@ def ingest_from_csv(
+        ...     ),
+        ...     time_array_id_columns=["id"],
+        ... )
+        >>> store.ingest_table(df, schema)
+        >>> store.map_table_time_config(
+        ...     "devices_by_representative_time",
+        ...     TableSchema(
+        ...         name="devices_by_datetime",
+        ...         value_column="value",
+        ...         time_config=DatetimeRange(
+        ...             time_column="timestamp",
+        ...             start=datetime(2020, 1, 1, 0),
+        ...             length=8784,
+        ...             resolution=timedelta(hours=1),
+        ...         ),
+        ...         time_array_id_columns=["id"],
+        ...     ),
+        ... )
         """
+        src_schema = self._schema_mgr.get_schema(src_name)
         map_time(self._engine, self._metadata, src_schema, dst_schema)
+        with self._engine.connect() as conn:
+            self._schema_mgr.add_schema(conn, dst_schema)
+            conn.commit()
+            self._schema_mgr.update_cache(dst_schema)
 
-    def read_query(self, query: Selectable | str, config: TimeBaseModel) -> pd.DataFrame:
-        """Return the query result as a pandas DataFrame."""
-        with self._engine.begin() as conn:
-            return read_database(query, conn, config)
+    def read_query(
+        self,
+        name: str,
+        query: Selectable | str,
+        params: Any = None,
+        connection: Optional[Connection] = None,
+    ) -> pd.DataFrame:
+        """Return the query result as a pandas DataFrame.
+
+        Parameters
+        ----------
+        name
+            Table or view name
+        query
+            SQL query expressed as a string or salqlchemy Selectable
+        params
+            Parameters for SQL query if expressed as a string
+
+        Examples
+        --------
+        >>> df = store.read_query("SELECT * FROM devices")
+        >>> df = store.read_query("SELECT * FROM devices WHERE id = ?", params=(3,))
 
-    def read_table(self, schema: TableSchema) -> pd.DataFrame:
+        >>> from sqlalchemy import select
+        >>> table = store.schemas.get_table("devices")
+        >>> df = store.read_query(select(table).where(table.c.id == 3)
+        """
+        schema = self._schema_mgr.get_schema(name)
+        if connection is None:
+            with self._engine.begin() as conn:
+                return read_database(query, conn, schema.time_config, params=params)
+        else:
+            return read_database(query, connection, schema.time_config, params=params)
+
+    def read_table(self, name: str, connection: Optional[Connection] = None) -> pd.DataFrame:
         """Return the table as a pandas DataFrame."""
-        return self.read_query(f"SELECT * FROM {schema.name}", schema.time_config)
+        table = self.get_table(name)
+        stmt = select(table)
+        return self.read_query(name, stmt, connection=connection)
+
+    def read_raw_query(
```

Comment:
Note this new method. Some users will prefer it because it doesn't require the table schema. However, the user will not get timestamp conversion with SQLite.